### ISSUE TITLE
Unable to provide custom parameters

### ISSUE THREAD

**Issue by GustavBjordal** [MiniZinc/MiniZincIDE#113]
Under extra configuration parameters I am trying to add a custom parameter that is not specified in any .msc file.
So for example, using gecode, I try to add the custom parameter `-foo` of type `string` with value `bar`.
I then get the following error message 
```
minizinc: Unrecognized option or bad format `-foo'
minizinc: MiniZinc driver.
Usage: minizinc  [<options>] [-I <include path>] <model>.mzn [<data>.dzn ...] or just <flat>.fzn
More info with "/Applications/MiniZincIDE.app/Contents/Resources/minizinc --help"
```

Which looks like MiniZinc, not gecode, is complaining that the flag does not exist. I have also tried variations like `foo` and `--foo`.

Maybe I am misunderstanding something, but previously one could just provide arbitrary flags to the solvers, which was a feature I used quite often for experimental features/flags. Now this functionality seems to be completely removed?

The only workaround I can see right now is to add the flags to the solvers .msc file (but these are not available for the builtin?), at which point MiniZinc seems to recognize then, but then what is the purpose of the "add custom parameter" feature?

**Comment by zayenz** [comment]
I had the same problem when trying to use the workaround suggested in #112, see my comment in testing various combinations there.

**Comment by cyderize** [comment]
Since the options are now passed to `minizinc` rather than to the solver binary directly, arbitrary parameters would have to be wrapped in `--fzn-flags` similarly to when running `minizinc` on the command line.

So adding a parameter called `fzn-flags` of type `string` with value `-foo bar` should work correctly.

Of course, adding the parameter as an extra flag in the solver `.msc` file would also let you add in the parameter - if it supports the flag, it should probably just be added there.

**Comment by zayenz** [comment]
Given that providing flags using --fzn-flags is not that discoverable (in this case, the sample size of me and @GustavBjordal is one indicator of that), would it be possible to have an option to indicate if a flag should be passed to the minizinc driver or to the flatzinc solver? That should both be discorverable, and give an inidcation to users what is possible.

**Comment by GustavBjordal** [comment]
I agree with @zayenz. Actually I expected the custom flag to be automatically wrapped in `--fzn-flags` as from the context it seems like the custom parameter is for the solver rather than for MiniZinc.

Also I initially wanted to test using `--fzn-flags` before I posted this issue but had forgotten the name of the flag and found that `minizinc` no longer lists it when under `-help`! 
`./minizinc --help | grep "fzn-flags"  ` comes out empty for me.

**Comment by cyderize** [comment]
The --fzn-flags option is part of the FlatZinc solver backend so it only comes up in `minizinc --help org.minizinc.mzn-fzn`.

The problem with the IDE automatically wrapping parameters in `--fzn-flags` is that it would then have to know what kind of solver backend is being used, as only FlatZinc solvers use `--fzn-flags` whereas MiniZinc solvers `--mzn-flags`. And then other solvers like the MIP solvers sometimes have underlying solvers to pass arguments to (like CBC, which has `--cbcArgs`) but others don't.

I don't really want to have to make the IDE aware of the various quirks of the different solver backends if I can help it (or maybe we need a more unified way of specifying underlying solver flags in `minizinc`). Is there a reason why the relevant parameters can't be added as extra flags in the solver `.msc` file? I guess we don't have a way to add those from the IDE preferences window, so maybe that is something to look into.

**Comment by zayenz** [comment]
https://github.com/MiniZinc/MiniZincIDE/commit/f84076bb9d8eaaf8fff5fdfd89535e5979f8b2d3 adding `--backend-flags` seems like a great solution to this problem!

**Comment by cyderize** [comment]
This has now been fixed in version 2.5.1
